File: DSA/DakshSethi_590012025/Day4Question2.py
user_input = input("enter the numbers \n")
nums = list(map(int, user_input.strip().split()))

k = int(input("enter the number of rotations \n"))

#rotate
def rotate(nums, k):
    n = len(nums)
    k = k % n
    nums[:] = nums[-k:] + nums[:-k]
rotate(nums, k)
print("rotated list:", nums)

File: DSA/ShatakshiMishra_590015008/Day4Question2.cpp
#include <iostream>
using namespace std;

int main() {
    int n, k;
    cin >> n;

    int nums[n];
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    cin >> k;

    k = k % n;

    int rotated[n];

    for (int i = 0; i < n; i++) {
        rotated[(i + k) % n] = nums[i];
    }

    for (int i = 0; i < n; i++) {
        cout << rotated[i] << " ";
    }
}
